agent/engine/docker_task_engine_linux_test.go

```diff
@@ -155,16 +157,21 @@ func TestDeleteTask(t *testing.T) {
 
 	mockControl := mock_control.NewMockControl(ctrl)
 	cgroupResource := cgroup.NewCgroupResource("", mockControl, nil, "cgroupRoot", "", specs.LinuxResources{})
+	asmSecretResource := asmsecret.NewASMSecretResource("", nil, "executionId", true, nil, nil)
```

Comment:
We need to assert that`resource.Cleanup` method is being invoked for this resource, otherwise someone could remove that call altogether and nothing would prevent that to get to prod.

I think you could pass a secret via `SetCachedSecretValue` and then assert that `resource.GetCachedSecretValue` returns false after delete task. Or might be easier to make this resource a mock and assert that `Cleanup` is invoked.

Reply:
Good point, Updated it. I think making this resource a mock will need more changes: make resource and resource mock to be implementing a same interface, so just use getCachedSecretValue to assure cleanup is called.